### PR TITLE
Remove rollup script row limits

### PR DESCRIPTION
**Pull request by rshipp**
Remove the limits on row count during insert in the `create_rollup_tables` script. This script runs weekly on Friday mornings, and has been failing since at least 2023-04-07. Since we already batch the inserts, we think it's safe to remove these limits.

I'm adding a few teams to the reviewers on this, mostly just for awareness, but feel free to push back if you think this change is unsafe.

## Links

- [slack thread on this approach](https://codedotorg.slack.com/archives/C03CK49G9/p1683665914631939)
- jira ticket: [TEACH-392](https://codedotorg.atlassian.net/browse/TEACH-392)

## Followup task

Send some of these queries to a replica database, to reduce the load on production.

jira: [TEACH-501](https://codedotorg.atlassian.net/browse/TEACH-501)

### CODE REVIEW

**Comment by dmcavoy** [comment]
@rshipp just a heads up that @megcrenshaw is the person on our team who has looked into this area most recently. They are out this week and next week. I don't know how pressing this is but if its not pressing it might be worth getting Meg's input

**Comment by pablo-code-org** [comment]
@dmcavoy I agree, I added @megcrenshaw  as a reviewer for this.
Also not sure how removing this constrain will make the script run. is this coming from an exception in the script?

**Comment by davidsbailey** [comment]
do we know how long this script takes to run currently? I don't love the idea of reading an arbitrarily large amount of data into memory, but if we think that in practice this will just lead to the script running a little slower then that could be ok.

**Comment by rshipp** [comment]
here's the exception: https://app.honeybadger.io/projects/45435/faults/95065140. I'm not sure how long the script took to run when it was working. We could increase the limits instead of removing them, though I'd expect that'd put us back in the same broken state later. This PR doesn't change what is loaded into memory, since the exception occurs after the data was already loaded.

**Comment by davidsbailey** [comment]
I'm fine with infra owning the decision on whether reading this data without a size limit is going to be a problem for our production daemon machine.

> This PR doesn't change what is loaded into memory, since the exception occurs after the data was already loaded.

True, but it's not really fair to compare this PR to the option of leaving the cronjob running in a broken state indefinitely. the purpose of the error is to alert us to the new larger size of the data and respond accordingly, either by raising the limit, removing the limit, or fixing the code to batch the reads as well as the writes as done for other DB access later in the file.

**Comment by rshipp** [comment]
For what it's worth, the exception lines were added in 2018, and the write-batching was added in 2021. Based on your original comment in the jira ticket, I'd been assuming that these checks were less important now that we have that batching.

The urgency here is coming from RED team, so if they're fine with waiting, we can try some different solutions here. I'd defer to @dju90 on how/when we'd be able to prioritize that, if teacher tools continues to own the fix.

**Comment by dju90** [comment]
Reading through the comments here, I take it that increasing the limit would be the least risky and fastest fix for now. Is that correct? May is when RED needs to report org-level metrics, so I'd favor the limit adjustment for now if that's faster + if we're not confident in removing the limit entirely.

**Comment by pablo-code-org** [comment]
Let me high light this PR on the infra channel and add it to our discussion today at 11:30am

Like @davidsbailey said, I am concern about just increasing this arbitrarily. I would try to debug the current size and adjust the limit based on that. 
I think the biggest concern would be overloading the server in terms of memory, but I'll see if there are other concerns from the infra team.

To summarize MY concerns:
- Memory going wild on the server
- Not following up on this with the right row limits in place.

On the other hand, I also support quick solutions in order to get the data needed for the RED team, but let me talk to the team.

**Comment by bethanyaconnor** [comment]
Just want to note that I know Meg has done a lot of work on contact rollups in the past few months. They might have also worked on this script (in particular, it looks like it processes `pd_applications_status_logs`) but this is a different process than the contact rollups process.

File: bin/cron/create_rollup_tables
Comment (bethanyaconnor) [inline bin/cron/create_rollup_tables:148]: It looks like the error was being raised on line 147. I wonder if we can keep these other warnings in for now if we're concerned?

In particular, I think we hold all levels in memory during seeding (ie [here](https://github.com/code-dot-org/code-dot-org/blob/d2dffe78c33a1c969547fcb6fea0ace61726783d/dashboard/lib/tasks/seed.rake#L236) for DSLDefined levels)) so I think we'd start to see problems earlier than this if we had too many levels. Too many level sources is a different issue though as they're user generated

**Comment by pablo-code-org** [comment]
Hi @rshipp,

After talking to the infra team. here are some thoughts
1) Removing the exception just because it was thrown, is not good practice. But we definitely understand the need to unblock this.
2) If you are going to remove it, remove just the exception that is affecting you, but add a logger to track the number of rows inserted (I believe we have a logger for contact rollups that can be re used, and if not, I am happy to guide you here). Once we have those numbers, we need to follow up and bring the exception code back. 

3) Given the size of this, we don't think this will cause an issue, but the infra team is here if needed. This runs at midnight tonight.


I am going to reject, so we can go back and add the logging to track the number of rows so we can re - enable this guard code next week.

**Review by pablo-code-org — CHANGES_REQUESTED** [review]
Added comment with why I am rejecting 
- Adding logger & remove only the exception blocking the process
- Follow up on this and re enable the check once we know the right number of processed rows

**Comment by rshipp** [comment]
Is it safe to run this script on production outside of the scheduled cron job? otherwise, removing one exception at a time could mean we have to wait 3 weeks for the script to run, if all the exceptions end up raising.

**Comment by pablo-code-org** [comment]
I just read the code, and it doesn't seem to time aware when running, only for scheduling, meaning it is fine to run it outside of the regular schedule.
@sureshc  mentioned this script doesn't cause an issue for the database load, so I would safe it is fine.

Remove the first exception, and also print the 
- `contained_levels.size`
- `contained_levels_answers.size`
- `level_sources_multi_types.size`
That way you will know if there will be issue on the next lines. TBH I would add them to the logger so we can see the increase over time and adjust this in the future and even alert to be more proactive at fixing this

**Pull request closed without merging**
